eox_nelp/api_clients/__init__.py

```diff
@@ -0,0 +1,123 @@
+            }
+
+            cache.set(key, headers, response.get("expires_in", 300))
+
+        self.session = requests.Session()
+        self.session.headers.update(headers)
+
+    def _authenticate(self, client_id, client_secret):
+        """This method uses the session attribute to perform a POS request based on the
+        base_url attribute and the given path, if the response has a status code 200
+        this will return the json from that response otherwise this will return an empty dictionary.
+
+        Args:
+            client_id<str>: Public application identifier.
+            client_secret<str>: Confidential identifier.
+
+        Return:
+            Dictionary: Response from authentication request.
+        """
+        client = BackendApplicationClient(client_id=client_id)
+        oauth = OAuth2Session(client_id=client_id, client=client)
+        authenticate_url = f"{self.base_url}/oauth/token"
+
+        return oauth.fetch_token(
+            token_url=authenticate_url,
+            client_secret=client_secret,
+            include_client_id=True,
+        )
+
+    def make_post(self, path, data):
+        """This method uses the session attribute to perform a POST request based on the
+        base_url attribute and the given path, if the response has a status code 200
+        this will return the json from that response otherwise this will return an empty dictionary.
+
+        Args:
+            path<str>: makes reference to the url path.
+            data<Dict>: request body as dictionary.
+
+        Return:
+            Dictionary: Empty dictionary or json response.
+        """
+        url = f"{self.base_url}/{path}"
+
+        response = self.session.post(url=url, json=data)
+
+        if response.status_code == status.HTTP_200_OK:
+            return response.json()
+
+        LOGGER.error(
+            "An error has occurred trying to make post request to %s with status code %s",
+            url,
+            response.status_code,
+        )
+
+        return {}
+
+    def make_get(self, path, payload):
+        """This method uses the session attribute to perform a GET request based on the
+        base_url attribute and the given path, if the response has a status code 200
+        this will return the json from that response otherwise this will return an empty dictionary.
+
+        Args:
+            path<str>: makes reference to the url path.
+            payload<Dict>: queryparams as dictionary.
+
+        Return:
+            Dictionary: Empty dictionary or json response.
+        """
+        url = f"{self.base_url}/{path}"
+
+        response = self.session.get(url=url, params=payload)
+
+        if response.status_code == status.HTTP_200_OK:
+            return response.json()
+
+        LOGGER.error(
+            "An error has occurred trying to make a get request to %s with status code %s",
+            url,
+            response.status_code,
+        )
+
+        return {}
```

Comment:
I know this is a refactor, but here you could a message of error or something related to the response...
How can I know if it is a 200  empty or an error? By the way, is the opportunity to handle it or maybe in the future because is working with that way now.